src/contracts/dpos.ts

```diff
@@ -0,0 +1,99 @@
+import BN from 'bn.js'
+import { Client } from '../client'
+import { Contract } from '../contract'
+import { Address } from '../address'
+import {
+  RegisterCandidateRequest,
+  VoteRequest,
+  ElectRequest,
+  ListCandidateRequest,
+  ListCandidateResponse,
+  Candidate,
+  UnregisterCandidateRequest,
+  ListWitnessesRequest,
+  ListWitnessesResponse,
+  Witness
+} from '../proto/dpos_pb'
+import { unmarshalBigUIntPB } from '../big-uint'
+
+export interface ICandidate {
+  address: Address
+  pubKey: Uint8Array
+}
+
+export interface IWitness {
+  pubKey: Uint8Array
+  voteTotal: BN
+  powerTotal: BN
+}
+
+export class DPOS extends Contract {
+  static async createAsync(client: Client, callerAddr: Address): Promise<DPOS> {
+    const contractAddr = await client.getContractAddressAsync('dpos')
+    if (!contractAddr) {
+      throw Error('Failed to resolve contract address')
+    }
+
+    return new DPOS({ contractAddr, callerAddr, client })
+  }
+
+  constructor(params: { contractAddr: Address; callerAddr: Address; client: Client }) {
+    super(params)
+  }
+
+  registerCandidateAsync(pubKey: Uint8Array) {
+    const registerCandidateReq = new RegisterCandidateRequest()
+    registerCandidateReq.setPubKey(pubKey)
+    this.callAsync<void>('RegisterCandidate', registerCandidateReq)
```

Comment:
Needs `return` and the function signature should have return type of `Promise<void>`... or whatever this returns.